pkg/prometheus/promcfg.go

```diff
@@ -2317,6 +2319,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 
 	if sc.Spec.RelabelConfigs != nil {
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
+	}
+
+	if shards != 1 {
+		alreadyHasShardLabel := false
+		if sc.Spec.RelabelConfigs != nil {
+			for _, relabelConfig := range sc.Spec.RelabelConfigs {
+				if relabelConfig.Action == "hashmod" {
+					level.Debug(cg.logger).Log("msg", "found existing hashmod relabeling rule, skipping", "relabeling", relabelConfig)
+					alreadyHasShardLabel = true
+				}
+			}
+		}
+		if !alreadyHasShardLabel {
+			relabelings = generateAddressShardingRelabelingRules(relabelings, shards)
+		}
+	}
+
+	if sc.Spec.RelabelConfigs != nil || shards != 1 {
```

Comment:
do we need a condition here? At this point, `relabelings` has at least 1 item.

Reply:
No, not if we add sharding relabels for non-sharded setups. Fixed.

Reply:
I think you removed the conditional you had above, which means we need the condition back 😅 

IMHO we should keep the conditional before we get here, i.e. on line 2326